tutorials/application_design/c++11/subscriber.cxx

```diff
@@ -0,0 +1,254 @@
+    std::mutex mutex;
+
+    dds::sub::cond::ReadCondition metrics_condition(
+            metrics_reader_,
+            dds::sub::status::DataState::any(),
+            [this]() { metrics_app(); });
+
+    dds::sub::cond::ReadCondition transit_condition(
+            transit_reader_,
+            dds::sub::status::DataState::any(),
+            [this]() { transit_app(); });
+
+    dds::core::cond::GuardCondition display_condition;
+    display_condition.extensions().handler(
+            [this, &display_condition]() { display_app(); });
+
+    std::thread display_thread([&display_condition, &mutex]() {
+        for (;;) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(500));
+            std::lock_guard<std::mutex> lock(mutex);
+            display_condition.extensions().trigger_value(true);
+        }
+    });
+
+    waitset.attach_condition(metrics_condition);
+    waitset.attach_condition(transit_condition);
+    waitset.attach_condition(display_condition);
+
+    for (;;) {
+        waitset.dispatch();
+        std::lock_guard<std::mutex> lock(mutex);
+        display_condition.extensions().trigger_value(false);
+    }
+}
+
+void SubscriberDashboard::display_app()
+{
+    std::stringstream ss;
+    auto now = std::chrono::system_clock::now();
+    ss << "[[ DASHBOARD: " << now.time_since_epoch().count() << " ]]\n";
+    {
+        auto online = online_vehicles();
+        ss << "Online vehicles: " << online.size() << "\n";
+        for (auto &item : online) {
+            ss << "- Vehicle " << item.vin << ":\n";
+            ss << "  Fuel updates: " << item.fuel_history.size() << "\n";
+            ss << "  Last known destination: "
+               << (item.current_destination
+                           ? utils::to_string(*item.current_destination)
+                           : "None")
+               << "\n";
+            ss << "  Last known fuel level: "
+               << (item.fuel_history.empty()
+                           ? "None"
+                           : utils::to_string(item.fuel_history.back()))
+               << "\n";
+        }
+    }
+    {
+        auto offline = offline_vehicles();
+        ss << "Offline vehicles: " << offline.size() << "\n";
+        for (auto &item : offline) {
+            ss << "- Vehicle " << item.vin << ":\n";
+            ss << "  Mean fuel consumption: "
+               << std::accumulate(
+                          item.fuel_history.begin(),
+                          item.fuel_history.end(),
+                          0.0)
+                            / item.fuel_history.size()
+               << "\n";
+            ss << "  Known reached destinations: "
+               << item.reached_destinations.size() << "\n";
+            for (auto &destination : item.reached_destinations) {
+                ss << "    - " << utils::to_string(destination) << "\n";
+            }
+        }
+    }
+
+    std::cout << ss.str() << std::endl;
+}
+
+void SubscriberDashboard::metrics_app()
```

Comment:
I would call metrics_app and transit_app display_vehicle_metrics() and display_vehicle_route()

Reply:
These functions don't "display" themselves, they update the internal state only.

Internally, I discussed about renaming them `do_metrics` so the name implies "an action". 
Do you think this is a good idea? I would need to update it for all other API bindings.

Reply:
update_metrics/update_vehicle_route?

Reply:
I won't change this in this PR. Given that we're planning a follow-up PR to evaluate bigger changes, we can make those then.